#! /bin/sh -e
## 90_can-2004-0412.dpatch by Siggy Brentrup <bsb@debian.org>
##
## $URL$
## $Id$
##
## DP: Mandrake security patch for CAN-2004-0412.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/tfheen/external/pkg-mailman/trunk/Mailman/Commands/cmd_password.py trunk/Mailman/Commands/cmd_password.py
--- /home/tfheen/external/pkg-mailman/trunk/Mailman/Commands/cmd_password.py	2004-08-18 15:16:30.000000000 +0200
+++ trunk/Mailman/Commands/cmd_password.py	2004-08-18 15:16:33.000000000 +0200
@@ -1,4 +1,4 @@
-# Copyright (C) 2002 by the Free Software Foundation, Inc.
+# Copyright (C) 2002-2004 by the Free Software Foundation, Inc.
 #
 # This program is free software; you can redistribute it and/or
 # modify it under the terms of the GNU General Public License
@@ -49,6 +49,8 @@
         if mlist.isMember(address):
             password = mlist.getMemberPassword(address)
             res.results.append(_('Your password is: %(password)s'))
+            # Prohibit multiple password retrievals.
+            return STOP
         else:
             listname = mlist.real_name
             res.results.append(
@@ -62,6 +64,8 @@
         if mlist.isMember(address):
             password = mlist.getMemberPassword(address)
             res.results.append(_('Your password is: %(password)s'))
+            # Prohibit multiple password retrievals.
+            return STOP
         else:
             listname = mlist.real_name
             res.results.append(
